docs/content/preview/yugabyte-voyager/migrate/assess-migration.md

```diff
@@ -0,0 +1,122 @@
+
+![Migration report](/images/migrate/assess-migration.jpg)
+
+## Generate a Migration Assessment Report
+
+1. [Install yb-voyager](../../install-yb-voyager/).
+1. Prepare the source database.
+
+    1. Create a new user, `ybvoyager` as follows:
+
+        ```sql
+        CREATE USER ybvoyager PASSWORD 'password';
+        ```
+
+    1. Grant necessary permissions to the `ybvoyager` user.
+
+        ```sql
+        /* Switch to the database that you want to migrate.*/
+        \c <database_name>
+
+        /* Grant the USAGE permission to the ybvoyager user on all schemas of the database.*/
+
+        SELECT 'GRANT USAGE ON SCHEMA ' || schema_name || ' TO ybvoyager;' FROM information_schema.schemata; \gexec
+
+        /* The above SELECT statement generates a list of GRANT USAGE statements which are then executed by psql because of the \gexec switch. The \gexec switch works for PostgreSQL v9.6 and later. For older versions, you'll have to manually execute the GRANT USAGE ON SCHEMA schema_name TO ybvoyager statement, for each schema in the source PostgreSQL database. */
+
+        /* Grant SELECT permission on all the tables. */
+
+        SELECT 'GRANT SELECT ON ALL TABLES IN SCHEMA ' || schema_name || ' TO ybvoyager;' FROM information_schema.schemata; \gex
+        ```
+
+1. Assess migration - Voyager supports two primary modes for conducting migration assessments, depending on your access to the source database as follows:
+
+    1. **With source database connectivity**: This mode requires direct connectivity to the source database from the client machine where voyager is installed. You initiate the assessment by executing the `assess-migration` command of `yb-voyager`. This command facilitates a live analysis by interacting directly with the source database, to gather metadata required for assessment. A sample command is as follows:
+
+        ```sh
+        yb-voyager assess-migration --source-db-type postgresql \
+        --source-db-host hostname --source-db-user ybvoyager \
+        --source-db-password password --source-db-name dbname \
+        --source-db-schema schema1,schema2 --export-dir /path/to/export/dir
+        ```
+
+    1. **Without source database connectivity**: In situations where direct access to the source database is restricted, there is an alternative approach. Voyager includes packages with scripts present at `/etc/yb-voyager/gather-assessment-metadata/postgresql`. You can perform the following steps with these scripts.
+
+        1. Copy the scripts to a machine which has access to the source database.
+        1. Run the `yb-voyager-pg-gather-assessment-metadata.sh` script by providing the source connection string, the schema names, and path to a directory where metadata will be saved. For example,
+
+            ```sh
+            /path/to/yb-voyager-pg-gather-assessment-metadata.sh 'postgresql://ybvoyager@host:port/dbname' 'schema1|schema2' '/path/to/assessment_metadata_dir'
```

Comment:
We are not mentioning IOPS interval here @priyanshi-yb @aishwarya24

Reply:
Hmm, can be added here:

@aishwarya24, update this step -
 1. Run the `yb-voyager-pg-gather-assessment-metadata.sh` script by providing the source connection string, the schema names, path to a directory where metadata will be saved, and an optional argument of an interval to capture the IOPS metadata of the source (in seconds with a default value of 120 seconds). For example,
```sh
/path/to/yb-voyager-pg-gather-assessment-metadata.sh 'postgresql://ybvoyager@host:port/dbname' 'schema1|schema2' '/path/to/assessment_metadata_dir' '60'
```